src/middleware.ts

```diff
@@ -158,14 +158,19 @@ type PersistOptions<S> = {
   /**
    * A function returning another (optional) function.
    * The main function will be called before the state rehydration.
-   * The returned function will be called after the state rehydration.
+   * The returned function will be called after the state rehydration or when an error occured.
    */
-  onRehydrateStorage?: (state: S) => ((state: S) => void) | void
+  onRehydrateStorage?: (state: S) => ((state?: S, error?: Error) => void) | void
   /**
    * If the stored state's version mismatch the one specified here, the storage will not be used.
    * This is useful when adding a breaking change to your store.
    */
   version?: number
+  /**
+   * A function to perform persisted state migration.
+   * This function will be called when persisted state versions mismatch with the one specified here.
+   */
+  migrate?: (persistedState: S, version: number) => S | Promise<S>
```

Comment:
`persistedState` might not match `S` if the versions doesn't match. You may type this using `any` since we can't really know what the stored state look like?

Reply:
You're right. Okay, I have fixed it.

I'm not sure whether there is something wrong with GitHub. The commit isn't shown in this PR even though it has been pushed to my forked repo ([link to commit](https://github.com/sumartoyo/zustand/commit/72cba0820e1f3d357e5bd058fa6991bfa1d5cc96)).

Reply:
Strange, what appends if you push something new? Can you try (for example) to update your branch and push the changes?

Reply:
Yay! That works!

Reply:
Great!